### PR TITLE
blueprints: use reconcile decorator instead of relying on function name prefix

### LINKED ISSUE #7852: blueprints: find another way to run reconcile than with dir()
Now looking at it again I think there's probably a better way to do this than using `dir()` and then `getattr()`, probably something using a decorator?

Pseudo code:

```
@self.reconcile
def do_something(self):
    pass
```

(Not really part of this PR tbh)

_Originally posted by @BeryJu in https://github.com/goauthentik/authentik/pull/7590#discussion_r1422312556_

### PR DESCRIPTION
**Pull request by BeryJu**
## Details

closes #7852
---

## Checklist

-   [x] Local tests pass (`ak test authentik/`)
-   [x] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

### CODE REVIEW

File: authentik/blueprints/apps.py [hunk 1]
```diff
@@ -61,7 +63,19 @@ def _reconcile(self, prefix: str) -> None:
             except (DatabaseError, ProgrammingError, InternalError) as exc:
                 self.logger.warning("Failed to run reconcile", name=name, exc=exc)
 
-    def reconcile_tenant(self) -> None:
+    @staticmethod
+    def reconcile_tenant(func: Callable):
+        """Mark a function to be called on startup (for each tenant)"""
+        setattr(func, "_authentik_managed_reconcile", ManagedAppConfig.RECONCILE_TENANT_CATEGORY)
```
Comment (rissson) [inline authentik/blueprints/apps.py:69]: If there are many reconcile functions for this category, would one get overridden?

Comment (BeryJu) [reply authentik/blueprints/apps.py:69]: Yeah you can only have one category per reconcile function but that also matches the current behaviour

Comment (rissson) [reply authentik/blueprints/apps.py:69]: Fair enough, I didn't look into the code too much

Comment (rissson) [reply authentik/blueprints/apps.py:69]: And if needed we can always turn it into a list in the future

**Review by rissson — APPROVED** [review]
I'll approve, but let's merge after 2024.2

**Pull request merged**